src/role_squadsiege.js

```diff
@@ -40,20 +40,22 @@ roles.squadsiege.preMove = function(creep, directions) {
 
   }
 
-  if (!creep.memory.initialized) {
-    Memory.squads[creep.memory.squad].siege[creep.id] = {};
-    creep.memory.initialized = true;
-  }
-  var squad = Memory.squads[creep.memory.squad];
-  if (squad.action == 'move') {
-    if (creep.room.name == squad.moveTarget) {
-      let nextExits = creep.room.find(creep.memory.routing.route[creep.memory.routing.routePos].exit);
-      let nextExit = nextExits[Math.floor(nextExits.length / 2)];
-      let range = creep.pos.getRangeTo(nextExit.x, nextExit.y);
-      if (range < 2) {
-        Memory.squads[creep.memory.squad].siege[creep.id].waiting = true;
-        creep.moveRandom();
-        return true;
+  if (creep.memory.squad) {
+    if (!creep.memory.initialized) {
+      Memory.squads[creep.memory.squad].siege[creep.id] = {};
+      creep.memory.initialized = true;
+    }
+    var squad = Memory.squads[creep.memory.squad];
+    if (squad.action == 'move') {
+      if (creep.room.name == squad.moveTarget) {
+        let nextExits = creep.room.find(creep.memory.routing.route[creep.memory.routing.routePos].exit);
+        let nextExit = nextExits[Math.floor(nextExits.length / 2)];
```

Comment:
The initial plan was, that the squadsiege waits ~2 fields before the exit. The path is calculated to exactly that exit, so in this case it should make sense. Any example where this is bad (in this case)?

I want to remove this anyway.